library/initial-content/src/main/java/org/cru/godtools/init/content/task/Tasks.kt

```diff
@@ -64,21 +66,25 @@ internal class Tasks @Inject constructor(
         }
     }
 
-    fun initSystemLanguages() {
+    suspend fun initSystemLanguages() {
         if (dao.getCursor(Query.select<Language>().where(LanguageTable.SQL_WHERE_ADDED)).count > 0) return
 
         // add device languages if we haven't added languages before
         if (dao.get(Query.select<Language>().where(LanguageTable.SQL_WHERE_ADDED).limit(1)).isEmpty()) {
-            LocaleUtils.getFallbacks(context.deviceLocale, Locale.ENGLISH).toList()
-                // add all device languages and fallbacks
-                .onEach { downloadManager.addLanguage(it) }
-                .asSequence()
-                // set the first available language as the primary language
-                .firstOrNull { dao.find<Language>(it) != null }?.let { settings.primaryLanguage = it }
-
-            // always add english and bundled languages
-            downloadManager.addLanguage(Locale.ENGLISH)
-            BuildConfig.BUNDLED_LANGUAGES.forEach { downloadManager.addLanguage(LocaleCompat.forLanguageTag(it)) }
+            coroutineScope {
```

Comment:
What is the scope of coroutines?

Reply:
`coroutineScope` indicates that any coroutines launched within the scope need to complete before the scope finishes